cpp/examples/sbml_to_memilio.cpp

```diff
@@ -0,0 +1,921 @@
+
+    return true;
+}
+
+/*
+    @brief Creates the parameters.h file
+
+    @param model The model to be processed
+    @param filename The name of the file to be processed
+
+    Extracts the filename using :cpp:func:`get_filename(const char* filename)`, converts it to lower case and creates a file with the resulting name using `boost::filesystem`. Then it creates one struct for every parameter in the model. It uses the value as returned by libsbml as default value. (This may be overwritten in the example.cpp file.)
+*/
+bool create_parameters(Model* model, const char* filename)
+{
+    std::string lowercase_name = boost::to_lower_copy<std::string>(get_filename(filename));
+    std::string uppercase_name = boost::to_upper_copy<std::string>(get_filename(filename));
+
+    size_t number_parameters = model->getListOfParameters()->size();
+
+    std::ofstream parameters;
+    parameters.open(lowercase_name + "/parameters.h", std::ios::out);
+
+    parameters << "#ifndef " << uppercase_name << "_PARAMETERS_H" << std::endl;
+    parameters << "#define " << uppercase_name << "_PARAMETERS_H" << std::endl;
+    parameters
+        << "\n#include \"memilio/epidemiology/age_group.h\"\n#include "
+           "\"memilio/epidemiology/uncertain_matrix.h\"\n#include \"memilio/utils/custom_index_array.h\"\n#include "
+           "\"memilio/utils/parameter_set.h\"\n#include \"memilio/utils/uncertain_value.h\"\n"
+        << std::endl;
+    parameters << "namespace mio\n{" << std::endl;
+    parameters << "namespace " << lowercase_name << "\n{" << std::endl;
+
+    std::string parameterset_initializer = "using ParametersBase = ParameterSet<";
+
+    if (number_parameters == 0) {
+        parameters << "template <typename FP = ScalarType>\nstruct DummyParam {\n    using Type = double;\n    static "
+                      "Type get_default()\n    {\n        return 0.0;\n    }\n    static std::string name()\n    {\n   "
+                      "     return \"DummyParameter\";\n    }\n};\n"
+                   << std::endl;
+        parameterset_initializer = parameterset_initializer + "DummyParam<FP>, ";
+    }
+    else {
+        for (size_t i = 0; i < number_parameters; i++) {
+            Parameter param  = *(Parameter*)model->getListOfParameters()->get(i);
+            double value     = param.getValue();
+            std::string name = param.getId();
+            parameters << "template <typename FP = ScalarType>" << std::endl;
+            parameters << "struct " << name << " {\n    using Type = double;\n  static Type get_default()\n{"
+                       << std::endl;
+            parameters << "return " << value << ";\n}" << std::endl;
+            parameters << "static std::string name()\n{\n   return \"" << name << "\";\n}\n};\n" << std::endl;
+            parameterset_initializer = parameterset_initializer + name + "<FP>, ";
+        }
+    }
+    parameterset_initializer.pop_back();
+    parameterset_initializer.pop_back();
```

Comment:
I would prefer rewriting the loop logic, maybe by adding something like
```
if (i != 0) { parameterset_initializer += ", "; }
```
at the start of the loop.